dynamic/clientset/dynamic_clientset.go

```diff
@@ -33,70 +33,71 @@ import (
 type Clientset struct {
 	config    rest.Config
 	resources *dynamicdiscovery.ResourceMap
+	dc        dynamic.Interface
 }
 
-func New(config *rest.Config, resources *dynamicdiscovery.ResourceMap) *Clientset {
+func New(config *rest.Config, resources *dynamicdiscovery.ResourceMap) (*Clientset, error) {
+	dc, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("can't create dynamic client when creating clientset: %v", err)
+	}
 	return &Clientset{
 		config:    *config,
 		resources: resources,
-	}
+		dc:        dc,
+	}, nil
 }
 
 func (cs *Clientset) HasSynced() bool {
 	return cs.resources.HasSynced()
 }
 
-func (cs *Clientset) Resource(apiVersion, resource, namespace string) (*ResourceClient, error) {
+func (cs *Clientset) Resource(apiVersion, resource string) (*ResourceClient, error) {
 	// Look up the requested resource in discovery.
 	apiResource := cs.resources.Get(apiVersion, resource)
 	if apiResource == nil {
 		return nil, fmt.Errorf("discovery: can't find resource %s in apiVersion %s", resource, apiVersion)
 	}
-	return cs.resource(apiResource, namespace)
+	return cs.resource(apiResource), nil
 }
 
-func (cs *Clientset) Kind(apiVersion, kind, namespace string) (*ResourceClient, error) {
+func (cs *Clientset) Kind(apiVersion, kind string) (*ResourceClient, error) {
 	// Look up the requested resource in discovery.
 	apiResource := cs.resources.GetKind(apiVersion, kind)
 	if apiResource == nil {
 		return nil, fmt.Errorf("discovery: can't find kind %s in apiVersion %s", kind, apiVersion)
 	}
-	return cs.resource(apiResource, namespace)
+	return cs.resource(apiResource), nil
 }
 
-func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource, namespace string) (*ResourceClient, error) {
-	// Create dynamic client for this apiVersion/resource.
-	gv := apiResource.GroupVersion()
-	config := cs.config
-	config.GroupVersion = &gv
-	if gv.Group != "" {
-		config.APIPath = "/apis"
-	}
-	dc, err := dynamic.NewClient(&config)
-	if err != nil {
-		return nil, fmt.Errorf("can't create dynamic client for resource %v in apiVersion %v: %v", apiResource.Name, apiResource.APIVersion, err)
-	}
+func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource) *ResourceClient {
 	return &ResourceClient{
-		ResourceInterface: dc.Resource(&apiResource.APIResource, namespace),
-		dc:                dc,
-		gv:                gv,
-		resource:          apiResource,
-	}, nil
+		NamespaceableResourceInterface: cs.dc.Resource(apiResource.GroupVersionResource()),
+		gv:                             apiResource.GroupVersion(),
```

Comment:
I don't remember why I originally stored `gv` in addition to `resource`, since you can get one from the other. Can you try removing `gv` from the struct and see if that simplifies things without adding some other cost?

Reply:
Noted, will try removing `gv`.